Week-3/exercise_3_2/index.js

```diff
@@ -0,0 +1,24 @@
+const car = {
+  brand: 'Tata',
+  model: 'Nexon',
+  registrationNumber: 'BH12345',
+
+  displayDetails: function (ownerName) {
+    console.log(`${ownerName}, this is your car: ${this.registrationNumber}, ${this.brand} - ${this.model}`);
+  }
+};
+
+const myCarDetails = car.displayDetails.bind(car, 'Sameer');
+myCarDetails();
+
+const carCreatedOn = function (ownerName, date) {
+  console.log(`${this.brand} - ${this.model} owned by ${ownerName} was created on ${date}`);
+};
+
+carCreatedOn.apply(car, ['Sameer', '2019-01-01']);
+
+const carDetails = function (type) {
+  console.log(`${this.brand} - ${this.model} is an ${type} car`);
+};
+
+carDetails.call(car, 'Electric');
```

Comment:
add new line

Reply:
@Saurabhrkp can u also explain pros and cons of each

Reply:
In JavaScript, bind, call, and apply are all methods that can be used to change the this context of a function, and allow you to call a function with a specific this value. Here are some pros and cons of each method:

bind:

Pros:
It returns a new function with a fixed this context, which can be useful for creating functions with a specific this value that can be called later.
It allows you to pre-set some of the arguments for the function, so they don't need to be passed in when the function is called.
Cons:
It doesn't call the function immediately, so it's less useful for functions that need to be called right away.
call:

Pros:
It allows you to change the this context of a function and call it immediately, which can be useful for functions that need to be called right away.
It allows you to pass in arguments individually, which can be useful for functions with a variable number of arguments.
Cons:
It doesn't return a new function with a fixed this context, so you'll need to call it with the correct this value every time you want to use it.
apply:

Pros:
Like call, it allows you to change the this context of a function and call it immediately.
It allows you to pass in arguments as an array, which can be useful for functions with a variable number of arguments.
Cons:
Like call, it doesn't return a new function with a fixed this context, so you'll need to call it with the correct this value every time you want to use it.
In summary, bind is useful for creating reusable functions with a fixed this context, call and apply are useful for changing the this context of a function and calling it immediately. It depends on the use case and requirement of the function whether to use bind, call or apply.